internal/common/rules/rulematrix.go

```diff
@@ -4,10 +4,27 @@ import (
 	"gonum.org/v1/gonum/mat"
 )
 
+// LinkTypeOption gives an enumerated type for the various link types available for rules
+type LinkTypeOption int
+
+const (
+	ParentFailAutoRulePass LinkTypeOption = iota
+	NoLink
+)
+
+// RuleLink provides a containerised package for all linked rules
+type RuleLink struct {
+	Linked     bool
+	LinkType   LinkTypeOption
+	LinkedRule string
```

Comment:
Should this contain a RuleMatrix instead? How would you go about changing the linked rule? Do we still have a rulesInPlay matrix and if so, do both of the linked and the mother rule need to be in play for it to be evaluated properly?

Reply:
I think we should keep it as a string reference to the rule name, to ensure minimal regression impacts on voting and evaluation. Yes both the linked and mother rule needs to be in play, this makes sense right, we can vote to modify or remove either independent from each other. We don't need to modify any of our procedures but we still get the intended effect.